cassovary-core/src/main/scala/com/twitter/cassovary/util/io/AdjacencyListGraphReader.scala

```diff
@@ -73,7 +73,7 @@ class AdjacencyListGraphReader[T] (
    * @param filename Name of file to read from
    */
   private class OneShardReader(filename: String, nodeNumberer: NodeNumberer[T])
-    extends Iterator[NodeIdEdgesMaxId] {
+    extends Iterable[NodeIdEdgesMaxId] {
 
     private val outEdgePattern = ("""^(\w+)""" + separator + """(\d+)""").r
     var lastLineParsed = 0
```

Comment:
move lastLineParsed, lines, holder to inside the new Iterator block. That's why you are seeing empty iterator exception.